File: apps/frontend/src/app/domain/entities/ScrapedGameInfo.ts
export interface ScrapedGameInfo {
  externalId: number;
  name: string;
  summary: string | null;
  coverUrl: string | null;
  rating: number | null;
  releaseYear: number | null;
  genres: string[];
  platforms: string[];
  storeLinks: StoreLink[];
  playtime: PlaytimeInfo | null;
  source: string;
}

export interface StoreLink {
  storeName: string;
  url: string;
  storeId: string | null;
}

export interface PlaytimeInfo {
  mainStoryHours: number | null;
  mainPlusExtrasHours: number | null;
  completionistHours: number | null;
}

export interface GameSearchResult {
  query: string;
  results: ScrapedGameInfo[];
  source: string;
}

export interface EnrichedGameInfo {
  gameInfo: ScrapedGameInfo;
  inLibrary: boolean;
  catalogGameId: string | null;
  matchReason: string | null;
}

export interface EnrichedSearchResult {
  query: string;
  results: EnrichedGameInfo[];
  source: string;
}

export interface AddGameResult {
  success: boolean;
  canonicalGameId: string;
  created: boolean;
  message: string;
}

export interface ScraperStatus {
  enabled: boolean;
  source: string;
  message: string;
}

export interface RescrapeRequest {
  igdbId?: number;
}

export interface RescrapeResult {
  success: boolean;
  gameId: string;
  gameName: string;
  message: string | null;
  updatedFields: RescrapeUpdatedFields | null;
}

export interface RescrapeUpdatedFields {
  thumbnailUrl: string | null;
  steamAppId: number | null;
  steamLink: string | null;
  gogLink: string | null;
  epicLink: string | null;
  rating: number | null;
  genres: string[];
}

File: apps/api/src/main/java/com/robertforpresent/api/scraper/domain/model/ScrapedGameInfo.java
package com.robertforpresent.api.scraper.domain.model;

import org.jspecify.annotations.Nullable;

import java.util.List;

/**
 * Represents game information scraped from external sources like IGDB.
 * This is a read-only value object containing scraped data.
 */
public record ScrapedGameInfo(
        long externalId,
        String name,
        @Nullable String summary,
        @Nullable String coverUrl,
        @Nullable Double rating,
        @Nullable Integer releaseYear,
        List<String> genres,
        List<String> platforms,
        List<StoreLink> storeLinks,
        @Nullable PlaytimeInfo playtime,
        String source
) {
    public ScrapedGameInfo {
        genres = genres != null ? List.copyOf(genres) : List.of();
        platforms = platforms != null ? List.copyOf(platforms) : List.of();
        storeLinks = storeLinks != null ? List.copyOf(storeLinks) : List.of();
    }

    public record StoreLink(
            String storeName,
            String url,
            @Nullable String storeId
    ) {}

    public record PlaytimeInfo(
            @Nullable Double mainStoryHours,
            @Nullable Double mainPlusExtrasHours,
            @Nullable Double completionistHours
    ) {}
}
